scanner/decisions/0004-scannerv4-read-vulneralbilities.md

```diff
@@ -0,0 +1,41 @@
+# 0004 - Scanner V4 NVD CVSS enrichment data updating pipeline
+
+- **Author(s):** Yi Li
+- **Created:** [2023-09-11]
+
+## Context
+
+We are currently transitioning the StackRox Scanner to use ClairCore as its primary scanning engine. Our goals are twofold:
+
+1. Enable Central to retrieve NVD CVSS data and consolidate all CVSS data that contains available CVSS v3 scores.
+2. Ensure seamless integration between Scanner V4 and Central, for enriching vulnerabilities with CVSS data.
+
+Throughout this workflow, a new CVSS Updater equipped by a novel enricher in Central will download and consolidate the CVSS data, then creating a JSON bundle. 
+
+ClairCore will be instrumental in scanner V4 data retriever (will be explained in detail below) while requesting this CVSS data bundle from central, handling tasks such as downloading, parsing, and updating CVSS data for Scanner V4.
+
+## Decision
+
+The CVSS Updater will be housed within Central, where it will retrieve CVSS data from Google storage when needed. It is designed to make a single HTTP call for each yearly NVD data bundle, with the earliest data tracing back to 2002. The updater will also compare the sha256 from the CVSS meta file for each year to determine if any data can be bypassed.
+
+This Updater will consolidate the data that only contains valid CVSS v3 scores, generate a json file and store it as a zip file in Central's file system. This compressed file is typically around 50 MB in size.
```

Comment:
Per our talk today, I will suggest not to say Central's "file system", instead saying in Central's storage etc so that you can have the room on decision where you really want to store the file.

Reply:
@daynewlee I don't see an immediate need for storing the NVD security data in persistent storage. If Central restarts (deployed, crashed, etc.), having it wiped out is OK because we would pull the data from Google Storage as soon as Central is ready. There is also the fact that it consumes less memory. And I think we can move to persistent later on if needed.

Nonetheless, Central still has Offline Mode. We should leave the door open to support that. We should not do anything right now that could prevent the customer from updating the feed manually. I don't think using `blob.Snapshot` would be a problem, but worth mentioning.